src/popover_link.coffee

```diff
@@ -9,15 +9,12 @@ Ember.Widgets.PopoverLinkComponent = Ember.Component.extend
   popoverClassNames: []
   rootElement: '.ember-application'
   fade: yes
+  _popover: null
 
-  willDestroy: ->
+  willDestroyElement: ->
+    @get('_popover')?.destroy()
```

Comment:
Yes, good catch. We should probably also check if `_popover` is already destroyed here.